go/libraries/doltcore/sqle/dprocedures/dolt_checkout_helpers.go

```diff
@@ -78,8 +79,22 @@ func MoveWorkingSetToBranch(ctx *sql.Context, brName string, force bool) error {
 		if err != nil {
 			return err
 		}
-		if actions.CheckoutWouldStompWorkingSetChanges(currentRoots, newBranchRoots) {
-			return actions.ErrWorkingSetsOnBothBranches
+		if !isNewBranch && actions.CheckoutWouldStompWorkingSetChanges(currentRoots, newBranchRoots) {
```

Comment:
The idea here is fine but it seems like this logic should be encapsulated by `actions.CheckoutWouldStompWorkingSetChanges()`

Reply:
Fixed.